drivers/storage/portworx/components_test.go

```diff
@@ -2794,6 +2803,128 @@ func TestSecurityInstall(t *testing.T) {
+			ClusterIP: sdkServerIP,
+			Ports: []v1.ServicePort{
+				{
+					Name: pxutil.PortworxSDKPortName,
+					Port: int32(sdkServerPort),
+				},
+			},
+		},
+	})
+	driver := portworx{}
+	driver.Init(k8sClient, runtime.NewScheme(), record.NewFakeRecorder(10))
+	cluster := &corev1alpha1.StorageCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "px-cluster",
+			Namespace: "kube-test",
+		},
+		Spec: corev1alpha1.StorageClusterSpec{
+			Security: &corev1alpha1.SecuritySpec{
+				Enabled: true,
+				Auth: &corev1alpha1.AuthSpec{
+					GuestAccess: &corev1alpha1.GuestRoleManaged,
+					SelfSigned: &corev1alpha1.SelfSignedSpec{
+						TokenLifetime: &metav1.Duration{
+							// Since we have a token expiration buffer of one minute,
+							// a new token will constantly be fetched.
+							Duration: time.Second * 1,
+						},
+					},
+				},
+			},
+		},
+	}
+	// Initial run
+	setSecuritySpecDefaults(cluster)
+	securityComponent, ok := component.Get(component.SecurityComponentName)
+	require.True(t, ok)
+	component.DeregisterAllComponents()
+	component.Register(component.SecurityComponentName, securityComponent)
+	securityComponent.Initialize(k8sClient, hversion.Version{}, runtime.NewScheme(), record.NewFakeRecorder(10))
+	driver.UpdateStorageClusterStatus(cluster)
+	require.NoError(t, err)
+
+	err = driver.PreInstall(cluster)
+	require.NoError(t, err)
+
+	// Disable GuestAccess. Should expect an update to be called.
+	cluster.Spec.Security.Auth.GuestAccess = &corev1alpha1.GuestRoleDisabled
+	expectedRole := component.GuestRoleEnabled
+	expectedRoleInspectResp := &osdapi.SdkRoleInspectResponse{
+		Role: &expectedRole,
+	}
+	mockRoleServer.EXPECT().
+		Inspect(gomock.Any(), &osdapi.SdkRoleInspectRequest{
+			Name: component.SecuritySystemGuestRoleName,
+		}).
+		Return(expectedRoleInspectResp, nil).
+		Times(1)
+	mockRoleServer.EXPECT().
+		Update(gomock.Any(), gomock.Any()).
+		Return(&osdapi.SdkRoleUpdateResponse{
+			Role: &component.GuestRoleDisabled,
+		}, nil).
+		Times(1)
+	err = driver.PreInstall(cluster)
+	require.NoError(t, err)
+
+	// Enable guest access, should be updated again
+	cluster.Spec.Security.Auth.GuestAccess = &corev1alpha1.GuestRoleEnabled
+	expectedRole = component.GuestRoleEnabled
+	expectedRoleInspectResp = &osdapi.SdkRoleInspectResponse{
+		Role: &expectedRole,
+	}
+	mockRoleServer.EXPECT().
+		Inspect(gomock.Any(), &osdapi.SdkRoleInspectRequest{
+			Name: component.SecuritySystemGuestRoleName,
+		}).
+		Return(expectedRoleInspectResp, nil).
+		Times(1)
+	mockRoleServer.EXPECT().
+		Update(gomock.Any(), gomock.Any()).
+		Return(&osdapi.SdkRoleUpdateResponse{
+			Role: &component.GuestRoleEnabled,
+		}, nil).
+		Times(1)
+	err = driver.PreInstall(cluster)
+	require.NoError(t, err)
+
+	// set to managed to avoid more calls without a corresponding mock expect
+	cluster.Spec.Security.Auth.GuestAccess = &corev1alpha1.GuestRoleManaged
+	err = driver.PreInstall(cluster)
+	require.NoError(t, err)
+	// no expected mock calls
```

Comment:
Can we add a test case for invalid GuestRole and see if an event is raised for it?

Reply:
Added this